### PR TITLE
executor: fix data inconsistency after `load data ... replace into ...` | tidb-test=e5f82b826aeb7e912c2d4dc36389960daac36d96

### LINKED ISSUE #56408: Data inconsistency in non clustered PK after `load data ...replace into ...`
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```
➜  ~ cat /Users/fanzhou/a.csv
1|aa|beijing
1|aa|beijing
1|aa|beijing
1|aa|beijing
2|bb|shanghai
2|bb|shanghai
2|bb|shanghai
3|cc|guangzhou
```

```
mysql> create table a(id int, name varchar(20), addr varchar(100), primary key (id) nonclustered);
Query OK, 0 rows affected (0.02 sec)

mysql> load data local infile '/Users/fanzhou/a.csv' replace into table a fields terminated by '|' escaped by '' lines terminated by '\n';
Query OK, 13 rows affected (0.01 sec)
Records: 8  Deleted: 0  Skipped: 0  Warnings: 0

mysql> select count(*) from a;
+----------+
| count(*) |
+----------+
|        3 |
+----------+
1 row in set (0.01 sec)

mysql> select count(*) from a use index();
+----------+
| count(*) |
+----------+
|        8 |
+----------+
1 row in set (0.01 sec)

mysql> admin check table a;
ERROR 8223 (HY000): data inconsistency in table: a, index: PRIMARY, handle: 3, index-values:"" != record-values:"handle: 3, values: [KindInt64 1]"
```

### 2. What did you expect to see? (Required)
Data consistent, No error
### 3. What did you see instead (Required)
Data Inconsistency
### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by fzzf678**
This is an cherry-pick of #56415

### What problem does this PR solve?

Issue Number: close #56408

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by CbcWestwolf — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

File: executor/insert_common.go [hunk 1]
```diff
@@ -1201,10 +1201,13 @@ func (e *InsertValues) batchCheckAndInsert(ctx context.Context, rows [][]types.D
 					if handle == nil {
 						continue
 					}
-					_, err = e.removeRow(ctx, txn, handle, r, true)
+					skip, err = e.removeRow(ctx, txn, handle, r, true)
 					if err != nil {
 						return err
 					}
+					if skip {
+						break
```
Comment (lance6716) [inline executor/insert_common.go:1209]: `break` will skip `removeRow` for rest `r.uniqueKeys`. I'm not sure if this is expected. Please find some execution reviewers to check

**Comment by lance6716** [comment]
/unhold

I see this is the same logic in master branch

**Pull request merged**